test/features/events.js

```diff
@@ -0,0 +1,312 @@
+      gender: 'f',
+      nationality: 'gbr',
+      centre: 'bigone'
+    };
+
+    var detaineeAttrs = {
+      person_id: 123,
+      cid_id: 1234,
+      centre: 1,
+      gender: 'male',
+      operation: 'check in',
+      timestamp: detaineeUpdateDateString
+    }
+
+    var getSearchResult = () => Detainee.find({
+      centre: detaineeAttrs.centre,
+      person_id: detaineeAttrs.person_id
+    });
+
+    Given('a Detainee already exists with person id `' + detaineeAttrs.person_id + '` from centre `' + detaineeAttrs.centre + '`', () =>
+
+      global.initializeBarrelsFixtures().then(() =>
+          Detainee.create(detaineeAttrs).then(() =>
+              getSearchResult()
+                .then((models) => {
+                  expect(models).to.have.length(1);
+                  expect(models[0].gender).to.equal(detaineeAttrs.gender);
+                  return expect(models[0].cid_id).to.equal(detaineeAttrs.cid_id);
+                })
+            )
+        )
+    );
+
+    And('the time of the event timestamp (' + payload.timestamp + ') is later than the existing Detainee creation timestamp (' + detaineeAttrs.timestamp + ')', () =>
+      expect(payload.timestamp).to.be.above(detaineeAttrs.timestamp)
+    );
+
+    When('a valid detainee event from centre `' + payload.centre + '` with person id `' + payload.person_id + '` occurs', () =>
+      request(sails.hooks.http.app)
+        .post('/irc_entry/event')
+        .send(payload)
+        .expect(201)
+    );
+
+    Then('the existing detainee is updated from the check in event received', () =>
+      getSearchResult().then((models) => {
+        expect(models).to.have.length(1);
+        expect(models[0].gender).to.equal(Detainee.normalizeGender(payload.gender));
+        expect(models[0].cid_id).to.equal(payload.cid_id);
+      })
+    );
+
+  });
+
+  Scenario('Existing Detainee Should Not Be Updated From Check In Operation That Occurred Before The Detainee Creation', () => {
+
+    before(function () {
+      global.testConfig.initializeBarrelsFixtures = false;
+    });
+
+    after(function () {
+      global.testConfig.initializeBarrelsFixtures = true;
+    });
+
+    var eventUpdateDateString = (new Date(10000)).toISOString();
+    var detaineeUpdateDateString = (new Date(20000)).toISOString();
+
+    var payload = {
+      person_id: 123,
+      cid_id: 4321,
+      operation: 'check in',
+      timestamp: eventUpdateDateString,
+      gender: 'f',
+      nationality: 'gbr',
+      centre: 'bigone'
+    };
+
+    var detaineeAttrs = {
+      person_id: 123,
+      cid_id: 1234,
+      centre: 1,
+      gender: 'male',
+      operation: 'check in',
+      timestamp: detaineeUpdateDateString
+    }
+
+    var getSearchResult = () => Detainee.find({
+      centre: detaineeAttrs.centre,
+      person_id: detaineeAttrs.person_id
+    });
+
+    Given('a Detainee with person id `' + detaineeAttrs.person_id + '` from centre `' + detaineeAttrs.centre + '` already exists', () =>
```

Comment:
if you're going to include params like centre names and person ids and timestamps in the given please be consistent, they're wrapped with a mix of backticks, brackets and not at all in this file.